crates/evm/evm/src/executors/invariant/mod.rs

```diff
@@ -384,8 +390,66 @@ impl<'a> InvariantExecutor<'a> {
+                if let Some(ref mut x) = call_result.edge_coverage {
+                    if x.len() > 0 {
+                        let mut new_coverage = false;
+
+                        // Iterate over the current map and the history map together and update
+                        // the history map, if we discover some new coverage, report true
+                        x.iter_mut()
+                            // Use zip to add history map to the iterator, now we get tuple back
+                            .zip(self.history_map.borrow_mut().iter_mut())
+                            // For the tuple pair
+                            .for_each(|(curr, hist)| {
+                                // If we got a hitcount of at least 1
+                                if *curr > 0 {
+                                    // Convert hitcount into bucket count
+                                    // Convert hitcount into bucket count
+                                    let bucket = match *curr {
+                                        0 => 0,
+                                        1 => 1,
+                                        2 => 2,
+                                        3 => 4,
+                                        4..=7 => 8,
+                                        8..=15 => 16,
+                                        16..=31 => 32,
+                                        32..=127 => 64,
+                                        128..=255 => 128,
+                                    };
+
+                                    // If the old record for this edge pair is lower, update
+                                    if *hist < bucket {
+                                        *hist = bucket;
+                                        new_coverage = true;
+                                    }
+
+                                    // Zero out the current map for next fuzzing iteration
+                                    *curr = 0;
+                                }
+                            });
+
+                        if new_coverage {
+                            if let Some(corpus_dir) = &self.config.corpus_dir {
+                                let timestamp = time::SystemTime::now()
+                                    .duration_since(time::UNIX_EPOCH)
+                                    .expect("Time went backwards")
+                                    .as_secs()
+                                    .to_string();
+                                if let Err(err) = foundry_common::fs::create_dir_all(corpus_dir) {
+                                    error!(%err, "Failed to create invariant corpus dir");
+                                } else if let Err(err) = foundry_common::fs::write_json_file(
+                                    corpus_dir.join(timestamp).as_path(),
+                                    &current_run.inputs.clone(),
```

Comment:
Ideally we'd have the ABI sig here and not have to call `from_invariant_call` all of the time to recover it